consensus/safety-rules/src/process_client_wrapper.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{safety_rules_manager, ConsensusState, Error, SafetyRulesManager, TSafetyRules};
+use consensus_types::{
+    block::Block,
+    block_data::BlockData,
+    common::{Payload, Round},
+    quorum_cert::QuorumCert,
+    timeout::Timeout,
+    vote::Vote,
+    vote_proposal::VoteProposal,
+};
+use libra_config::{
+    config::{ConsensusType, NodeConfig, RemoteService, SafetyRulesBackend, SafetyRulesService},
+    utils,
+};
+use libra_types::crypto_proxies::{Signature, ValidatorSigner};
+use std::{
+    any::TypeId,
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+};
+
+/// This container exists only so that we can kill the spawned process after testing is complete.
+/// Otherwise the process will be killed at the end of the safety_rules function and the test will
+/// fail.
+pub struct ProcessClientWrapper<T> {
+    signer: ValidatorSigner,
+    _safety_rules_manager: SafetyRulesManager<T>,
+    safety_rules: Box<dyn TSafetyRules<T>>,
+}
+
+impl<T: Payload> ProcessClientWrapper<T> {
+    pub fn new(backend: SafetyRulesBackend) -> Self {
+        let server_port = utils::get_available_port();
+        let server_address = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), server_port);
+
+        let type_id = TypeId::of::<T>();
+        let consensus_type = if type_id == TypeId::of::<Round>() {
```

Comment:
hmm, there's probably way to pass the generics all the way down to avoid reflection but probably not worth in this pr

Reply:
We have to translate this into a serializable type because we're starting a process to communicate this protocol. Though maybe I missed your point.